ui-admin/src/study/participants/RequestKitModal.tsx

```diff
@@ -84,3 +87,21 @@ export const useKitTypeSelect = (studyEnvParams: StudyEnvParams) => {
       onChange={option => setKitType(option?.value ?? '')}/>
   }
 }
+
+/**
+ *
+ */
+export const useBadAddressOverride = (initialValue: boolean) => {
```

Comment:
Not sure I'm a huge fan of this approach over a component - I could be convinced, but the added simplicity doesn't seem worth the cost of confusion.

Reply:
interesting -- to me these hooks that let the control manage its own state are less confusing since they handle everything themselves.  I'll add it to the eng. strategy.  I know there are cases where a component is best, and where a hook like this is best (e.g. the useReactTable hook), we could use some good guidance on when to use which

Reply:
Hmm, I see what you're saying. I see hooks as being `set up internal state (that has no connection to DOM)` and components as `connect thing to your DOM`. I think `useReactTable` is less confusing because it only returns what you're supposed to add to the dom - but at that point, I feel like it should just be a component. Idk.